services/src/main/java/org/keycloak/protocol/oidc/grants/OAuth2GrantTypeBase.java

```diff
@@ -244,7 +249,9 @@ protected String getRequestedScopes() {
 
         if (!validScopes) {
             event.error(Errors.INVALID_REQUEST);
-            throw new CorsErrorResponseException(cors, OAuthErrorException.INVALID_SCOPE, "Invalid scopes: " + scope,
+            String errorMessage = "Invalid scopes: " + scope;
```

Comment:
Same concern like above (the fact that we're returning scopes inside the CorsErrorResponseException is likely also far from ideal, so if you can change in the PR when on this, it can be nice :-)

Reply:
Same as above. I'll remove the received scopes from the returned response. How about logging a warning here with the invalid scope?

Reply:
Thanks. Also sounds good to me to log the warning. Sending back the valid scopes is also a good idea.

Reply:
I updated the code, the error response now looks like this:
```bash
curl -s \
  -d "grant_type=password" \
  -d "username=admin" \
  -d "password=admin" \
  -d "client_id=admin-cli" \
  -d "scope=unknown" \
  http://localhost:8081/auth/realms/master/protocol/openid-connect/token 
```

```json
{"error":"invalid_scope","error_description":"Invalid scopes. Scopes must be contained in [address, email, microprofile-jwt, offline_access, organization, phone, profile]"}
```